### PR TITLE
New blog post: Online DDL: why FOREIGN KEYs are not supported

### PR DESCRIPTION
**Pull request by shlomi-noach**
This is a cross-post of http://code.openark.org/blog/mysql/the-problem-with-mysql-foreign-key-constraints-in-online-schema-changes, with some language/grammar changes.

The purpose of the cross post is to have a Vitess-owned URL to which we can point users asking about `FOREIGN KEY` support, or lack thereof, in Online DDL.

cc @deepthi @askdba

### CODE REVIEW

**Comment by shlomi-noach** [comment]
This is the blog post preview: https://deploy-preview-774--vitess.netlify.app/blog/2021-06-15-online-ddl-why-no-fk/

**Comment by shlomi-noach** [comment]
ping

File: content/en/blog/2021-06-15-online-ddl-why-no-fk.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+If `person` made it, and `company` does not - if `company`'s migration breaks/fails for whatever reason - we're left in inconsistent and impossible scenario. `person` is committed to `company_ghost`, but `company` is still committed to `country`. We have to keep that lock on `country` and run a new migration on `company`! and again, and again. Meanwhile, `country` is locked. Meanwhile `person` is also locked. You can't write to `person` because you can't verify that related rows exist in `country`, because `country` has a `WRITE` lock.
+
+Can't stress this enough: the lock must not be released until all children tables are migrated. So, for our next challenge, what happens on a failover? We get referential integrity corruption, because locks don't work across servers.
+
+### Disk space
+
+Remember that an OSC works by creating a _ghost_ table and populating it until it is in sync with the original table. This effectively means requiring extra disk space at roughly the same volume as the original table.
+
+In a perfect world, we'd have all the disk space we ever needed. In reality we don't have that luxury. Sometimes we're not sure we have the space fo reven a single migration.
```
Comment (derekperkins) [inline content/en/blog/2021-06-15-online-ddl-why-no-fk.md]: s/fo reven/for even

**Review by derekperkins — APPROVED** [review]

**Review by askdba — APPROVED** [review]

**Comment by shlomi-noach** [comment]
Should have updated the date to June 22

**Pull request merged**